### PR TITLE
[DX] lock dev dependencies and remove Behat

### PR DESCRIPTION
**Pull request by oallain**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?   | no
| Fixed issue | 

In order to improve the developer experience, I lock the version of the dev dependencies. 🔒 
This allows not to have random errors in the CI at each PR because the dependencies have evolved. 🎉 
Same as https://github.com/synolia/SyliusSchedulerCommandPlugin/pull/70

I also deleted Behat because it was only used for one test that didn't really have any added value. I think it would be nice to test the backoffice with PHPUnit tests.

### CODE REVIEW

**Review by TheGrimmChester — DISMISSED** [review]

File: ruleset/phpstan.neon [hunk 1]
```diff
@@ -1,10 +1,11 @@
 parameters:
-    level: max
+    level: 8
```
Comment (Jibbarth) [inline ruleset/phpstan.neon:2]: I would suggest to you to leave `max` here, and fix errors or generate a baseline :wink:

File: ruleset/phpstan.neon [hunk 2]
```diff
@@ -24,3 +25,4 @@ parameters:
         - '#Cannot call method [a-zA-Z0-9]+\(\) on Symfony\\Component\\Config\\Definition\\Builder\\NodeParentInterface\|null.#'
         - '#Call to an undefined method League\\Pipeline\\PipelineInterface\:\:process\(\).#'
         - '#Method Doctrine\\Persistence\\ObjectRepository<mixed>\:\:findOneBy\(\) invoked with 2 parameters, 1 required.#'
+        - '#Method Doctrine\\Persistence\\ObjectRepository<object>\:\:findOneBy\(\) invoked with 2 parameters, 1 required.#'
```
Comment (Jibbarth) [inline ruleset/phpstan.neon:28]: maybe we should install `phpstan/doctrine-extension` ?

Comment (Jibbarth) [reply ruleset/phpstan.neon:28]: Is the line just above still valid ?

File: ruleset/phpstan.neon (hunk 1 above)
Comment (oallain) [reply ruleset/phpstan.neon]: Job done ;)

File: ruleset/phpstan.neon (hunk 2 above)
Comment (oallain) [reply ruleset/phpstan.neon:26]: I modified the previous line that was no longer valid rather than adding this one.

**Review by TheGrimmChester — APPROVED** [review]

File: ruleset/ecs.php [hunk 3]
```diff
@@ -4,16 +4,19 @@
...
     $parameters->set(Option::PATHS, [
         dirname(__DIR__, 1) . '/src',
-        dirname(__DIR__, 1) . '/tests/Behat',
         dirname(__DIR__, 1) . '/tests/PHPUnit',
-        dirname(__DIR__, 1) . '/spec',
+    ]);
+
+    $parameters->set(Option::SKIP, [
+        \PhpCsFixer\Fixer\Phpdoc\PhpdocTagTypeFixer::class,
+        // @TODO remove this line when we drop php7.4 support
```
Comment (Jibbarth) [inline ruleset/ecs.php:19]: Seen together, it's for fix a php8.1 build, so does it should really removed when drop php7.4 ?

**Review by Jibbarth — APPROVED** [review]

**Review by TheGrimmChester — APPROVED** [review]

**Pull request merged**